integration/kafka/src/main/java/alluxio/kafka/connect/AlluxioSinkTask.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    String alluxioUrl = mConfig.getString(AlluxioSinkConnectorConfig.ALLUXIO_URL);
+    String topDir = mConfig.getString(AlluxioSinkConnectorConfig.TOPICS_DIR);
+    String topicsUrl = alluxioUrl + "/" + topDir;
+    String tmpTopicsUrl = topicsUrl + "/tmp";
+
+    //create topics dir
+    AlluxioURI topicsDirPath = new AlluxioURI(topicsUrl);
+    try {
+      if (!mFs.exists(topicsDirPath)) {
+        mFs.createDirectory(topicsDirPath);
+      }
+      AlluxioURI tmpTopicsDirPath = new AlluxioURI(tmpTopicsUrl);
+      if (!mFs.exists(tmpTopicsDirPath)) {
+        mFs.createDirectory(tmpTopicsDirPath);
+      }
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to create " + topDir + " dir");
+    }
+
+    Collection<TopicPartition> assignments = context.assignment();
+    try {
+      initializeTopicPartitionWriter(assignments);
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to initialize TopicPartitionWriter");
+    }
+  }
+
+  /**
+   * Initialize AlluxioTopicPartitionWriter according to task assignments.
+   * One TopicPartition corresponds to one stream.
+   *
+   * @param assignments assigned TopicPartitions for this task
+   */
+  private void initializeTopicPartitionWriter(Collection<TopicPartition> assignments)
+    throws AlluxioException, IOException {
+    for (TopicPartition tp : assignments) {
+      AlluxioTopicPartitionWriter tpWriter =
+          new AlluxioTopicPartitionWriter(mConfig, context, mFormat, mFs, tp);
+      tpWriter.initialize();
+      mTpWriters.put(tp, tpWriter);
+    }
+  }
+
+  @Override
+  public void put(Collection<SinkRecord> records) {
+    for (SinkRecord record : records) {
+      String topic = record.topic();
+      int partition = record.kafkaPartition();
+      TopicPartition topicPartition = new TopicPartition(topic, partition);
+      AlluxioTopicPartitionWriter writer = mTpWriters.get(topicPartition);
+      if (writer != null) {
+        writer.writeToBuffer(record);
+      } else {
+        LOG.error("Failed to get writer of " + topicPartition.toString());
+      }
+    }
+
+    for (AlluxioTopicPartitionWriter writer : mTpWriters.values()) {
+      try {
+        writer.writeRecord();
+      } catch (IOException | AlluxioException e) {
+        LOG.error("Failed to write record to " + writer.getTopicPartition().toString());
+        LOG.error("Exception is {}", e);
+      }
+    }
+  }
+
+  /**
+   * Flushes all records that have been put for the specified topic-partitions. The
+   * offsets are provided for convenience, but could also be determined by tracking all offsets
+   * included in the SinkRecords passed to put method.
+   *
+   * @param offsets mapping of TopicPartition to committed offset
+   */
+  @Override
+  public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
```

Comment:
@Ishiihara thanks for pointing it out. Can you be more specific on how to implement this `flush` method? Is there any additional logic to be added?